pkg/cache/application.go

```diff
@@ -628,6 +647,22 @@ func (app *Application) handleReleaseAppAllocationEvent(event *fsm.Event) {
 			}
 		}
 	}
+	app.postAppReleased()
+}
+
+func (app *Application) postAppReleased() {
+	if app.GetApplicationState() != events.States().Application.Resuming {
+		return
+	}
+	canRunApplication := true
+	for _, task := range app.taskMap {
+		if task.GetTaskState() != events.States().Task.Completed {
+			canRunApplication = false
+		}
+	}
+	if canRunApplication {
+		dispatcher.Dispatch(NewRunApplicationEvent(app.applicationID))
+	}
```

Comment:
I think we miss one important check here (apologies did not find this out last time). 
When we check task states, we should only check the task that are placeholders. The second minor thing, I think we can remove bool var `canRunApplication` to simplify this function a bit. Something like this:

```
if app.GetApplicationState() != events.States().Application.Resuming {
  return
}	

for _, task := range app.taskMap {	
  if  task.placeholder && task.GetTaskState() != events.States().Task.Completed {
     return
  }
}

log.Logger().Info("Resuming completed, start to run the app",
  zap.String("appID", app.applicationID),
  zap.String("reason", eventArgs[0]))
dispatcher.Dispatch(NewRunApplicationEvent(app.applicationID))
```